packages/myst-cli/src/project/fromTOC.ts

```diff
@@ -236,13 +133,15 @@ export function patternsToFileEntries(
         }
         ignore.push(...listExplicitFiles(newEntries, path));
         return newEntries;
+      } else {
+        const { children } = entry as ParentEntry;
+        if (children) {
+          const newChildren = patternsToFileEntries(session, children, path, ignore, file);
+          return { ...entry, children: newChildren };
+        } else {
+          return entry;
+        }
```

Comment:
Make `children` a mutually exclusive property against `pattern`

Reply:
I believe this comment has been addressed under `myst-toc` in this PR!